src/Processor/CancelOrRefundProcessor.php

```diff
@@ -0,0 +1,57 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Webhook Module for PHP
+ *
+ * Copyright (c) 2021 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Webhook\Processor;
+
+use Adyen\Webhook\EventCodes;
+use Adyen\Webhook\PaymentStates;
+
+class CancelOrRefundProcessor extends Processor implements ProcessorInterface
+{
+    const MODIFICATION_ACTION = 'modification.action';
+    const CANCEL = 'cancel';
+    const REFUND = 'refund';
+
+    public function process(): ?string
+    {
+        $state = $this->initialState;
+        $logContext = [
+            'eventCode' => EventCodes::CANCEL_OR_REFUND,
+            'originalState' => $state
+        ];
+
+        if ($this->notification->isSuccess() && isset($this->notification->additionalData[self::MODIFICATION_ACTION])) {
```

Comment:
Will `CANCEL_OR_REFUND` always have a `modification.action`? Maybe we should default to either `cancel` or `refund` if there's no `modification.action`

Reply:
yes CANCEL_OR_REFUND always have `modification.action`.